runtime/compare.c

```diff
@@ -190,14 +228,30 @@ static intnat do_compare_val(struct compare_stack* stk,
         continue;
     }
     case String_tag: {
-      mlsize_t len1, len2;
+      mlsize_t len1, len2, len_remaining;
       int res;
+      const char *str1, *str2;
+      bool last_block;
```

Comment:
A small style comment again: the codebase now accepts declarations inside blocks, so it is better practice to declare variables at the moment of their initialization.